Makefile

```diff
@@ -1,10 +1,17 @@
+.PHONY: clean requirements upgrade
```

Comment:
Shouldn't `test` also be included in this list?

Reply:
yes.